src/plugins/discover/public/application/context/context_app_route.tsx

```diff
@@ -7,42 +7,36 @@
  */
 import React, { useEffect } from 'react';
 import { useParams } from 'react-router-dom';
-import { i18n } from '@kbn/i18n';
 import { EuiEmptyPrompt } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n-react';
 import { ContextApp } from './context_app';
-import { getRootBreadcrumbs } from '../../utils/breadcrumbs';
 import { LoadingIndicator } from '../../components/common/loading_indicator';
-import { useDataView } from '../../hooks/use_data_view';
-import { useMainRouteBreadcrumb } from '../../hooks/use_navigation_props';
-import { useDiscoverServices } from '../../hooks/use_discover_services';
+import { useDataView } from './hooks/use_data_view';
+import { getScopedHistory } from '../../kibana_services';
+import { ContextHistoryLocationState } from './services/locator';
 
 export interface ContextUrlParams {
   dataViewId: string;
   id: string;
 }
 
 export function ContextAppRoute() {
-  const services = useDiscoverServices();
-  const { chrome } = services;
+  const scopedHistory = getScopedHistory();
+  const [locationState, setLocationState] = React.useState<ContextHistoryLocationState>(
+    (scopedHistory.location.state as ContextHistoryLocationState) || {}
+  );
```

Comment:
Started to review and I'm wondering why we need the location state in a different way than in `discover_main_route.tsx` ? Where `useRef` is used

https://github.com/elastic/kibana/blob/d0002dc58f9535dcccfaa9cf42535886df96a48d/src/plugins/discover/public/application/main/discover_main_route.tsx#L66-L71

Reply:
Seems like it was needed for update id logic which was removed. Thx for catch.